docs/design_notes/module_lifecycle.md

```diff
@@ -0,0 +1,289 @@
+# The Module Lifecycle
+
+This design note is intended for people who are somewhat familiar with Modules but want to understand more about the design principles behind the abstraction. After reading this, you should have a solid understanding of the assumptions and guarantees the Module API is built upon. If you have no practical experience with Modules yet, Please take a look at the [MNIST tutorial](https://flax.readthedocs.io/en/latest/notebooks/annotated_mnist.html).
+
+Modules are the core of Flax. This is a practical guide to the design and behavior of the `Module` class. By the end, you should feel comfortable to go off the beaten track and use Modules in new ways.
```

Comment:
Suggestion:


```suggestion
Modules are at the core of Flax. This note is a practical guide about the design and behavior of the `Module` class. After studying this note, you should feel more comfortable with Modules and use them in custom ways suited for your tasks.
```

Note: https://developers.google.com/style/translation?hl=en#be-inclusive (writing for a global audience)